addons/calendar/models/calendar_recurrence.py

```diff
@@ -129,38 +129,46 @@ class RecurrenceRule(models.Model):
          "The day must be between 1 and 31"),
     ]
 
+    def get_recurrence_name(self):
+        rrule_type_selection = dict(self._fields['rrule_type']._description_selection(self.env))
+        weekday_selection = dict(self._fields['weekday']._description_selection(self.env))
+        byday_selection = dict(self._fields['byday']._description_selection(self.env))
+
+        period = rrule_type_selection[self.rrule_type]
+        every = _("Every %(count)s %(period)s", count=self.interval, period=period)
+
+        if self.end_type == 'count':
+            end = _("for %s events", self.count)
+        elif self.end_type == 'end_date':
+            end = _("until %s", self.until)
+        else:
+            end = ''
+
+        if self.rrule_type == 'weekly':
+            weekdays = self._get_week_days()
+            # Convert Weekday object
+            weekdays = [str(w) for w in weekdays]
+            # We need to get the day full name from its three first letters.
+            week_map = {v: k for k, v in RRULE_WEEKDAYS.items()}
+            weekday_short = [week_map[w] for w in weekdays]
+            day_strings = [weekday_selection[day] for day in weekday_short]
+            on = _("on %s") % ", ".join(day_strings)
+        elif self.rrule_type == 'monthly':
+            if self.month_by == 'day':
+                position_label = byday_selection[self.byday]
+                weekday_label = weekday_selection[self.weekday]
+                on = _("on the %(position)s %(weekday)s", position=position_label, weekday=weekday_label)
+            else:
+                on = _("day %s", self.day)
+        else:
+            on = ''
+
+        return ' '.join(filter(lambda s: s, [every, on, end]))
```

Comment:
This will be difficult, if not impossible, to translate in some languages that have a very different syntax from English. The reason for this is that you break the text you want to translate into smaller pieces that you translate individually and then reassemble in a defined order (here: `every`, `on`, `end`). But in some languages, the syntax of the sentence might be `end`, `on`, `every`, or `on`, `end`, `every`. The translator should be able to re-order that. Not to mention that strings like `"on %s"` can be complicated to translate without any more context: what is "on"? the antonym of "off"? or like, positioned *on* the top of something? or like in "*on* the 12th January"? Wiktionary gives no less than 59 meanings to the word "on"

Here are two pieces of advice that you can try to apply to improve the translatability of your code:
1. Never concatenate translated text. Don't try to factorize text, text is not code, don't be afraid to repeat yourself.
2. Always provide as much context as possible. This means complete sentences, with named variables.